### DISCUSSION TITLE
Added new feature but cannot figure out how to test :|

### DISCUSSION THREAD

**Discussion by fruitbatinshades** [riok/mapperly#1253] [Q&A]
I have a requirement to alter the property names being search for `dtmDOB` to `DOB`  or `MyName` to `My_Name` for instance. I followed the `PropertyNameMappingStrategy` config as I feel this is related (matching property names) I have added this functionality as a config setting with a Lambda

```csharp
    /// <summary>
    /// If normal matching did not find a match, change the property name via this method
    /// </summary>
    public Func<string, string> PropertyNameProcess { get; set; } // x=> x.Replace("_","")
```

**Q.** I can't quite figure out how to get that config set in the tests. The test below passes if I hard code the lambda but I can't figure out how to set it in the test itself

```csharp
public class ObjectFieldTest
{
    [Fact]
    public void OneSimpleField()
    {
        var source = TestSourceBuilder.Mapping(
            "A",
            "B",
            "class A { public string StringValue; }",
            "class B { public string String_Value; }"
        );

        TestHelper
            .GenerateMapper(source) //HOW DO I SET CONFIGURATION?
            .Should()
            .HaveSingleMethodBody(
                """
                var target = new global::B();
                target.String_Value = source.StringValue;
                return target;
                """
            );
    }
```

Code is in https://github.com/fruitbatinshades/mapperly if anyone wants to have a look or offer suggestions.

**Comment by latonz** [comment]
Thanks for considering contributing to Mapperly!

I'm not sure if I understand correctly what you are trying to implement. Please also keep in mind to open an issue for any major new feature to discuss the feature and its design before implementation to prevent redundant work. See also the [contributor documentation](https://mapperly.riok.app/docs/contributing/#missing-a-feature).

Note the `TestSourceBuilderOptions.Mapping` method has an overload that accepts `TestSourceBuilderOptions` which allows to set options how the test source code is built. The test source code is built in `TestSourceBuilder.MapperWithBody`, the `MapperAttribute` in `TestSourceBuilder.BuildAttribute`.

If I get it correctly you want to add a new property to the `MapperAttribute` named `PropertyNameProcess` of type `Func<string, string>`, is that correct? A similar feature is discussed in #854 (with a similar idea as you had) and #619. However, this is not really easy with source generators (see discussion in #854). The problem here is that the source generator does not have access to the compiled function `PropertyNameProcess` (only to its symbols and syntax). But to build the mappings / match the properties Mapperly would need to execute this function. This probably is possible by extracting this function from the user's code and compiling it in a separate context (see for example this [PoC](https://github.com/hermanussen/CompileTimeMethodExecutionGenerator)). However in the past we decided explicitly not to this, as this would introduce a lot of complexity to the Mapperly source and comes with a big performance hit.

**Reply by fruitbatinshades** [reply]
We make heavy use of an automapper feature but have cross framework libraries (net48, net8.0) and automapper is not currently compatible. (10 has a reflection bug, 11 does not support .net standard). So I am looking ata viable replacement. The features I still need are for mapping property names. automapper uses 

```
RecognizePrefixes(new[] { "str", "int", "dtm", "ysn", "Str", "Int", "Dtm", "Ysn" });
SourceMemberNamingConvention = new LowerUnderscoreNamingConvention();
DestinationMemberNamingConvention = new PascalCaseNamingConvention();
```

When we map our horrible db objects to the business layer and DTO's we "need" to strip those out, hence trying `PropertyNameProcess` as a way to implement those types of operations.

**Reply by latonz** [reply]
The way to go for this would be https://github.com/riok/mapperly/issues/619.
